app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -102,7 +103,13 @@ class ScanSimpleIdUpdate @Inject constructor(
     return if (event.patients.isEmpty()) {
       searchPatientOnlineWhenOnlinePatientLookupEnabled(event, model)
     } else {
-      next(model = model.notSearching(), patientFoundByIdentifierSearch(patients = event.patients, identifier = event.identifier))
+      when (model.openedFrom) {
+        OpenedFrom.EditPatientScreen.ToAddBpPassport, OpenedFrom.EditPatientScreen.ToAddNHID -> dispatch(ShowScannedQrCodeError(IdentifierAlreadyExists))
```

Comment:
Same as above comment, since we are not displaying specific messages based on if it's NHID or BP Passport, we can just go with checking if it's opened from `EditPatientScreen`